encoding/codecs/x264.md

```diff
@@ -0,0 +1,292 @@
+
+You don't need to use a tuning,
+but it generally helps
+to produce a better-looking video.
+
+#### `--crf 24`
+
+CRF is a constant-quality, 1-pass encoding mode.
+In layman's terms,
+this means that we don't need the output to meet a specific filesize,
+we just want the output to meet a certain quality level.
+CRF ranges from 0 to 51 (for 8-bit encoding),
+with 0 being the best quality
+and 51 being the smallest filesize,
+but there is a certain range of CRF settings
+that are generally most useful.
+Here are some guidelines:
+
+- CRF 13: This is considered visually lossless to videophiles.
+  This can produce rather large files,
+  but is a good choice if you want high quality videos.
+  Some fansubbing groups use this for Blu-ray encodes.
+- CRF 16-18: This is considered visually lossless to most viewers,
+  and leans toward high quality
+  while still providing a reasonable filesize.
+  This is a typical range for fansub encodes.
+- CRF 21-24: This provides a good balance between quality and filesize.
+  Some quality loss becomes visible,
+  but this is generally a good choice
+  where filesize becomes a concern,
+  such as for videos viewed over the internet.
+- CRF 26-30: This prioritizes filesize,
+  and quality loss becomes more obvious.
+  It is generally not recommended to go higher than CRF 30
+  in any real-world encoding scenario,
+  unless you want your videos to look like they were made for dial-up.
+
+#### `-o x264output.mkv -`
+
+This last portion tells which files to use for the input and output.
+We use `-o` to tell which filename to write the encoded file to.
+In this case, x264 will write a file at `x264output.mkv`
+in the current directory.
+
+The last argument we are passing to x264 is the input file.
+In this case, we pass `-` for the input file,
+which tells x264 to use the piped output from vspipe.
+The input argument is the only positional argument,
+so it does not need to be last;
+x264 will recognize it
+as the only argument without a `--` flag before it.
+
+## Example 2: Targeted File Size
+
+For the next example,
+let's say we want to make sure our encode
+fits onto a single 4.7GB DVD.
+How would we do that in x264?
+
+First, we'll need to figure out
+what bitrate our encode should be,
+in **kilobits per second**.
+This means we'll need to know a couple of things:
+
+- The length of our video, in seconds.
+  For this example,
+  let's say our movie is 2 hours (120 minutes) long.
+  We'll convert that to seconds:
+  120 minutes \* 60 minutes/second = **7200 seconds**.
+- Our target filesize.
+  We know that this is 4.7GB,
+  but we need to convert it to kilobits.
+  We can do this with the following steps:
+
+```
+4.7GiB * 1024MiB/GiB = 4812.8MiB
+4812.8MiB * 1024KiB/MiB = 4928307.2KiB
+4928307.2KiB * 8Kbits/KiB = 39426457.6Kbits
+```
+
+Now we divide the kilobit size we calculated by our video length,
+to find our kilobit per second target bitrate:
+
+```
+39426457.6Kbits / 7200 seconds = 5475 Kbps
+```
```

Comment:
We have the option of changing this math to render with TeX (which also might come in handy on other pages where math has been put in codeblocks or in-line currently) with the `katex` gitbook plug-in. Here's what the changes would look like (and the rendering works with the current text/theme options we give the user).

### Currently:
![image](https://user-images.githubusercontent.com/10972040/66485677-36a06580-ea77-11e9-919e-45b938ba8c2a.png)

### With katex:
![image](https://user-images.githubusercontent.com/10972040/66485731-49b33580-ea77-11e9-93e2-03daa6f3de05.png)
![image](https://user-images.githubusercontent.com/10972040/66485848-79623d80-ea77-11e9-9087-a080f850d0f8.png)

I know @FichteFoll has mentioned concerns about adding more plug-ins to the book, so I'll ask for their feedback on this.

---

With the current 12 plug-ins, initial generation: `finished with success in 2.0s`
"" on file-changed restart: `finished with success in 1.7s`

With katex, initial generation: `finished with success in 2.5s`
"" on file-changed restart: `finished with success in 1.9s`

Reply:
I love LateX. Let's do this! Thanks for taking the time to look into (as I considered suggesting this as well but due to being relatively short on time I didn't want to commit to it).